plugins/elasticsearch-bboss/src/main/java/com/navercorp/pinpoint/plugin/elasticsearchbboss/interceptor/ElasticsearchExecutorOperationInterceptor.java

```diff
@@ -0,0 +1,187 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.navercorp.pinpoint.plugin.elasticsearchbboss.interceptor;
+
+import com.navercorp.pinpoint.bootstrap.context.MethodDescriptor;
+import com.navercorp.pinpoint.bootstrap.context.SpanEventRecorder;
+import com.navercorp.pinpoint.bootstrap.context.TraceContext;
+import com.navercorp.pinpoint.bootstrap.interceptor.SpanEventSimpleAroundInterceptorForPlugin;
+import com.navercorp.pinpoint.common.trace.AnnotationKey;
+import com.navercorp.pinpoint.plugin.elasticsearchbboss.ElasticsearchConstants;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * @author yinbp[yin-bp@163.com]
+ */
+public class ElasticsearchExecutorOperationInterceptor extends SpanEventSimpleAroundInterceptorForPlugin {
+    private boolean recordResult = false;
+    private boolean recordArgs = false;
+    private boolean recordDsl = false;
+    private boolean recordResponseHandler = false;
+    private boolean recordESVersion = false;
+    private Method getClusterVersionInfo;
+
+    private int maxDslSize;
+
+    public ElasticsearchExecutorOperationInterceptor(TraceContext context, MethodDescriptor descriptor) {
+        super(context, descriptor);
+        recordResult = this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordResult",false);
+        recordArgs = this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordArgs",true);
+        recordDsl =  this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordDsl",true);
+        maxDslSize =  this.getTraceContext().getProfilerConfig().readInt("profiler.elasticsearchbboss.maxDslSize",ElasticsearchConstants.maxDslSize);
+        recordResponseHandler =  this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordResponseHandlerClass",false);
+        recordESVersion = this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordESVersion",true);
+
+
+    }
+    @Override
+    public void before(Object target, Object[] args) {
+        super.before(target,args);
+
+    }
+    @Override
+    public void after(Object target, Object[] args, Object result, Throwable throwable){
+
+
+        super.after(target,args,result,throwable);
+    }
+
+
+    @Override
+    protected void doInBeforeTrace(SpanEventRecorder recorder, Object target, Object[] args) {
+
+        recorder.recordServiceType(ElasticsearchConstants.ELASTICSEARCH_EXECUTOR);
+    }
+
+    private String getClusterVersionInfo(Object target){
+        try {
+            if(getClusterVersionInfo != null)
+                return (String)getClusterVersionInfo.invoke(target);
```

Comment:
How returning saved the clusterVersion(type `String`) here.

and save the String from below.

this way we don't have to invoke the method everytime.

Reply:
Each target(ClientInstance) has a specific version of Elasticsearch Datasource,and each Elasticsearch Datasource retains its corresponding Elasticsearch cluster version information such as Elasticsearch 1.x or 2.x or 5.x or 6.x or 7.x or 8.x and so on.
so we should get elasticsearchClusterVersionInfo in target everytime.

Reply:
I've understood. If that's the case, to avoid method invocation due to performance. 

How about considering creation(by instrumentation) of a variable named `clusterVersion`(type `String`) in the `ClientInstance`? 

In this way, setting with `_$PINPOINT$_setClusterVersion` in the interceptor for the first time. 
and retrieve with `_$PINPOINT$_getClusterVersion`.
(_$PINPOINT$_ is the prefix we use when instrumenting variables)

You can refer to `DataSourceMonitorAccessor.java` as an example.

Reply:
Good Idea，I'll do this.

Reply:
Modify Get clusterVersion code  as follow:
```java
 private String getClusterVersionInfo(Object target){
        try {
            ClusterVersionAccessor clusterVersionAccessor = (ClusterVersionAccessor)target;
            if(clusterVersionAccessor._$PINPOINT$_getClusterVersion() != null){
                return clusterVersionAccessor._$PINPOINT$_getClusterVersion();
            }
            else{
                synchronized (target.getClass()) {
                    if(clusterVersionAccessor._$PINPOINT$_getClusterVersion() != null){
                        return clusterVersionAccessor._$PINPOINT$_getClusterVersion();
                    }
                    Method _getClusterVersionInfo = target.getClass().getMethod("getClusterVersionInfo");
                    String version = (String) _getClusterVersionInfo.invoke(target);
                    if (version == null) {
                        clusterVersionAccessor._$PINPOINT$_setClusterVersion("UNKNOWN_VERSION");
                    } else {
                        clusterVersionAccessor._$PINPOINT$_setClusterVersion(version);
                    }
                    return clusterVersionAccessor._$PINPOINT$_getClusterVersion();
                }
            }
        } catch (Exception e) {

        }
        return "UNKNOWN_VERSION";

    }
```
inject pinpoint get/set clusterVersion field:
```java
public static class RestSearchExecutorTransformCallback implements TransformCallback{
		@Override
		public byte[] doInTransform(Instrumentor instrumentor, ClassLoader loader,
									String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain,
									byte[] classfileBuffer) throws InstrumentException {

			final InstrumentClass target = instrumentor.getInstrumentClass(loader, className, classfileBuffer);
			target.addField(ClusterVersionAccessor.class);
...//ommitted
```

Reply:
When we use instrumented variables, there exists a check statement as below

``` java
if (target instanceof DataSourceMonitorAccessor) {  <----- check statement
            final DataSourceMonitorAccessor dataSourceMonitorAccessor = (DataSourceMonitorAccessor) target;

            final DbcpDataSourceMonitor dataSourceMonitor = dataSourceMonitorAccessor._$PINPOINT$_getDataSourceMonitor();
```
what do you think about this?

Reply:
I think :
After injecting pinpoint get/set clusterVersion field to the target class by RestSearchExecutorTransformCallback ,the target must be an ClusterVersionAccessor object,so checking is not required unless the field was not injected successfully.

Reply:
I've added Type check